### ISSUE TITLE
14.0b{3,5} SDK is not usable with GCC bootstrap compiler.

### ISSUE THREAD

**Issue by iains** [iains/gcc-12-branch#7]
@fxcoudert - SDK issue is still present [not able to analyse right now] - would you be able to try this on 13 Beta?
(I will now try a later GCC as the bootstrap - but I'm guessing that the issue is something to do with Availability macros or similar)

Using gcc-7.5 as bootstrap on Intel Monterey with CLT 14.0b5:

```
x86_64-apple-darwin19-g++ -std=c++11 -O1 -pipe -fno-exceptions -fno-rtti -include config.h -I/src-local/gcc-git-12/libcody \
	  -MMD -MP -MF buffer.d -c -o buffer.o /src-local/gcc-git-12/libcody/buffer.cc
In file included from /src-local/gcc-git-12/libcody/cody.hh:33:0,
                 from /src-local/gcc-git-12/libcody/internal.hh:5,
                 from /src-local/gcc-git-12/libcody/buffer.cc:6:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/socket.h:425:57: error: expected constructor, destructor, or type conversion before ‘;’ token
 DECLARE_CONSTRAINED_PTR_TYPES(struct sockaddr, sockaddr);
                                                         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/socket.h:471:73: error: expected constructor, destructor, or type conversion before ‘;’ token
 DECLARE_CONSTRAINED_PTR_TYPES(struct sockaddr_storage, sockaddr_storage);
                                                                         ^
```

**Comment by iains** [comment]
same error with GCC-10.4 and GCC-12.2 as the bootstrap.

edit: I'd imagine that the problem would also show at stage2 if using clang as the bootstrap .. will try that tomorrow.

edit2: well, that test is hard to make since the stage1 library build fails with the linker error ... so we never get as far as stage 2 libcody.

**Comment by fxcoudert** [comment]
I see a lot of system headers use `DECLARE_CONSTRAINED_PTR_TYPES`
which is defined in `<sys/constrained_ctypes.h>`  but is wrapped with `#if defined(__clang__)`

On other compilers, `DECLARE_CONSTRAINED_PTR_TYPES` is therefore not defined, and the system headers simply can’t compile. Great.

**Comment by iains** [comment]
well, GCC is not the only non-clang compiler on macOS .. so let's hope that this can be fixed.

**Comment by fxcoudert** [comment]
The SDK header issue (`DECLARE_CONSTRAINED_PTR_TYPES`) is now known as FB11474414, and a fix is also being tested.

**Comment by iains** [comment]
Fixed in 14.1(RC and release) I think - there are other issues with the macOS 13 SDK (improper deprecation of C library functions) but those are described elsewhere.
